core/src/Kokkos_Complex.hpp

```diff
@@ -829,4 +829,13 @@ struct reduction_identity<Kokkos::complex<T> > {
 
 } // namespace Kokkos
 
+namespace std
+{
+  template <typename Number>
+  Number abs( Kokkos::complex<Number> n )
```

Comment:
I am pretty sure `abs` is not a customization point.  Also, the preexisting conversion from `Kokkos::complex<T>` to `std::complex<T>` should handle this.  Check out the following godbolt example: https://godbolt.org/z/PQiQD9
```c++
#include <complex>

template<class T>
class MyComplex {
public:
  MyComplex (const T& re, const T& im) : re_ (re), im_ (im) {}
  operator std::complex<T> () const { return {re_, im_}; }
private:
  T re_, im_;
};
int main() {
  MyComplex<double> z (2.0, 3.0);
  std::complex<double> zs = z;
  double a = std::abs (zs);
  return 0;
}
```

Reply:
Yes, specializing `std::abs` for a custom type should be fine (https://en.cppreference.com/w/cpp/language/extending_std), but not necessary here.